lib/puppet/provider/package/yum.rb

```diff
@@ -25,6 +25,14 @@
 
   defaultfor :osfamily => :redhat
 
+  def execute(*args)
```

Comment:
That could be more elegant, but I was only suggesting that a helper like `#execute` be moved to a shared location so all package providers can make use of it. Maybe rename it to something like `#execute_in_environment`.